### DISCUSSION TITLE
How to add to Home Assistant?

### DISCUSSION THREAD

**Discussion by Phoenix17785** [macxq/foxess-ha#217] [Q&A] [answered]
Hi,

I added HACS and added the needed information into the configuration.yaml with File Editor. 

How can I now see my 3 FoxESS Inverters in Home Assistant? 

I'm a Newbie with HA.

Just recently started with it.

Thanks in advance

**Comment by FozzieUK** [comment]
If you have added the three platform statements into your configuration.yaml for each of your inverters, you must then restart the HA (from the left menu Developer Tools, on the right hand side click 'RESTART' and select restart Home Assistant.

When it restarts it should then open connections to the FoxESS cloud for each of your inverters.

The list of sensors can be seen on this integrations [Readme page](https://github.com/macxq/foxess-ha?tab=readme-ov-file#-provided-entities) - if you have allocated a 'name' to each inverter you should be able to search for all entities with each name and add them as needed to your dashboard.

For example if your first inverter has a name of Fox1 and you click on the Overview page and in the search box at the top right of the screen enter Fox1 and it will show all the entities that are available for that inverter and you can build your dashboard up from there.

Have a look at the links at the bottom of the main page, there is some information on setting up and using sample sensors there.

**Reply by Phoenix17785** [reply]
Canb't restart as configuration.yaml moans about

bad indentation of a mapping entry (13:11)

 10 | script: .include scripts.yaml
 11 | scene: .include scenes.yaml
 12 | 
 13 |     sensor:
----------------^
 14 |   - platform: foxess
 15 |   

Any help is appreciated.

**Reply by FozzieUK** [reply]
The spacing is very strict in the configuration file you should have 2 spaces in front of the '- platform' so it lines up with the 'n' of sensor

Like this - 
```
sensor:
  - platform: foxess
    username: yourusername
    password: yourpassword
```

**Comment by Phoenix17785** [comment]
Any other recommendations for the File Editor?

still get an error

**Reply by FozzieUK** [reply]
i'd need to see more of your configuration file to know - for example you must only have one line that has 'sensor:' in it and there should be no spaces before it, the original error you posted implied there was something after sensor: as it's complaining about the 11th character - make sure there are no spaces after it as well.

If you can post the lines up to and including the foxess platform lines (but type over your username and password with random text)
Use the link in the editor '<>' to insert your copied code  or if you type 3 of these **```** characters to start the code block and another three to end it will preserve your formatting so I can see whats going on.

**Reply by Phoenix17785** [reply]
# Loads default set of integrations. Do not remove.
default_config:

# Load frontend themes from the themes folder
frontend:
  themes: !include_dir_merge_named themes

automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml

sensor:
  - platform: foxess
    name:     H3-8-0
    username: foxesscloud_username xxx
    password: foxesscloud_password xxx
    deviceID: foxesscloud_inverter_id xxx
  - platform: foxess
    name:     H3-10-0
    username: foxesscloud_username xxx
    password: foxesscloud_password xxx
    deviceID: foxesscloud_inverter_id xxx
  - platform: foxess
    name:     H3-10-0
    username: foxesscloud_username xxx
    password: foxesscloud_password xxx
    deviceID: foxesscloud_inverter_id xxx

no error message now ... saved ... restarted HA

But still I don't find the sensors?

![Bildschirmfoto 2024-02-14 um 19 46 45](https://github.com/macxq/foxess-ha/assets/155656072/e3c473be-e619-4c33-8515-8a5608b3504d)

**Reply by FozzieUK** [reply]
Ok, looks like the spacing errors are cleared 👍 

Apo9logies for my poor translation but 

If you click on 'Developer Tools (Entwicklerwerkseuge)' and click 'STATES' in the entity box type foxess - does it show any values ?

If not can you check 'Settings (Einstellungen), System, Logs' and post any logs that refer to platform foxess (note you might need to click 'Load Full Logs' to copy and paste any error messages)

**Reply by Phoenix17785** [reply]
![Bildschirmfoto 2024-02-15 um 08 52 32](https://github.com/macxq/foxess-ha/assets/155656072/7021466b-9f22-46b7-969b-1a72231f0f2e)

Nothing under States under Developer Tools

LogFiles

![Bildschirmfoto 2024-02-15 um 08 53 33](https://github.com/macxq/foxess-ha/assets/155656072/75641142-97c2-42ab-972e-7ec5c4756e18)

Did I upload those 3 files correct to custom_components? 

![Bildschirmfoto 2024-02-15 um 08 54 12](https://github.com/macxq/foxess-ha/assets/155656072/9f71bf28-9858-4d9e-bc58-c82922023acb)

**Reply by FozzieUK** [reply]
Ahh that looks to be the problem if you have manually installed - those files should be in this directory -
```
homeassistant/custom_components/foxess
```
Can you create the foxess directory and manually copy the __init__py, manifest.json and sensor.py files to it - then restart Home Assistant

**Comment by Phoenix17785** [comment]
Hi, 

says Wrong Username and Password? 

But when I type the same direcly under the FoxESS Cloud I can login???? 

![Bildschirmfoto 2024-02-15 um 10 33 58](https://github.com/macxq/foxess-ha/assets/155656072/d02b97b7-2b57-41a1-992b-d3e25ff68274)

![Bildschirmfoto 2024-02-15 um 10 34 27](https://github.com/macxq/foxess-ha/assets/155656072/db5aea65-3e14-4822-8bd9-058924043ab6)

![Bildschirmfoto 2024-02-15 um 10 35 35](https://github.com/macxq/foxess-ha/assets/155656072/d2b94d73-2094-4163-bc24-eb5aff112d94)


Thanks so much for your help so far !

**Reply by FozzieUK** [reply]
ok, we just had a very similar problem where the user had entered their username and password with the 'foxesscloud_' prefix (as per the readme)

That prefix isn't meant to be there it was just meant as use your 'foxesscloud' username/password (I will correct the readme to clarify that shortly)

So for example if your 

my_username = "Freddy42"
my_password = "FredsPass"
my_deviceID = c82393f3-7475-4f6b-ac86-123456789012

The platform entry would be (without foxesscloud_ in front):
```
  - platform: foxess
    username: Freddy42
    password: FredsPass
    deviceID: c82393f3-7475-4f6b-ac86-123456789012
```

If you could change that and restart HA

**Reply by Phoenix17785** [reply, marked as answer]
It seems it doesn't like the ! I have at the beginning of my password. I have another ! at the end which seems to be fine. 

So I need to edit my Password for FoxESSCloud Login it seems.

**Reply by FozzieUK** [reply]
ahh right, I had heard that it didn't like certain reserved characters but wasn't sure which ones - '!' needs to be added to the list

**Comment by Phoenix17785** [comment]
All three inverters are in plus Battery. 

Thank you so much. 

Can we please edit the readme accordingly so others can profit of my challenges I had? 

Thank you

**Reply by FozzieUK** [reply]
ok great, no problem i'll make the changes to Readme 👍 

You won't see it for a couple of weeks as there is a new version being tested - FoxESS are closing the old cloud api down for private access (that this uses) and we are transferring across to their newer 'openapi' so I will make the changes on that test branch. 
It is due to be released toward the end of this month once FoxESS have confirmed the timing.

**Reply by Phoenix17785** [reply]
Will will Users be able to use that new Integration?

Is that correct that I see everything in the Overview dashboard? Or can I also create a new one just for FoxESS and associate the entities for Battery etc. to it?

**Reply by FozzieUK** [reply]
Yes the new integration will work as it does now - the new openAPI needs an api_key adding to the setup which comes from your Foxesscloud account, it's easy enough to get and will be explained in the release notes.

Most users create a separate dashboard for the Foxess items as the overview can get cluttered.
You can create new dashboards in Settings, Dashboards and they can be auto-populated with all available entities at setup.

Once you have a Foxess dashboard I guess you would then create 3 separate cards, 1 for each of your inverters and then add the individual entities you are in interested in monitoring batter_soc, pv1/2 power etc.. 

You might want to create some sensors that 'sum' each inverters PV output so you can see the daily totals, this link has some good information about how you go about setting up sums and utility meters
[Sensors for Solar Monitoring](https://github.com/macxq/foxess-ha/wiki/Sample-sensors-for-better-solar-monitoring)
